src/hdmf/container.py

```diff
@@ -535,11 +550,26 @@ class Data(AbstractContainer):
     """
 
     @docval({'name': 'name', 'type': str, 'doc': 'the name of this container'},
-            {'name': 'data', 'type': ('scalar_data', 'array_data', 'data'), 'doc': 'the source of the data'})
+            {'name': 'data', 'type': ('scalar_data', 'array_data', 'data'), 'doc': 'the source of the data'},
+            {'name': 'term_set', 'type': TermSet, 'doc': 'the set of terms used to validate data on add',
+             'default': None})
     def __init__(self, **kwargs):
         data = popargs('data', kwargs)
+        self.term_set = popargs('term_set', kwargs)
         super().__init__(**kwargs)
-        self.__data = data
+        if self.term_set is not None:
+            bad_data = []
+            for term in data:
+                if self.validate_data(term=term, term_set=self.term_set):
+                    continue
```

Comment:
Similar to my comment above, if `term_set` is already a member of `Data` then as a user I would expect to be able to call `validate_data` with just the term, i.e., `validate_data` should use `self.term_set` by default. Also, if `term=None`, I think `validate_data` could just do the iteration over all terms directly.

Reply:
Data is validated when a new instance of Data is created and is not supported if the data already exists. However, add_ref_termset will return "missing terms" from the data if the term is invalid to the termset. The user never really has to call "validate_data" directly. The method is called in the presence of a term_set for VectorData (if it is provided) and also with add_ref_term_set. Since the user doesn't call it directly, I don't think the term_set parameter being explicit matters.

Reply:
On second thought, if the user never calls it directly, then the method could be private and use self.term_set in the method rather than "termset=self.termset" and setting the kwarg later on.